chargebee/src/main/java/com/chargebee/android/billingservice/BillingClientManager.kt

```diff
@@ -287,69 +421,135 @@ class BillingClientManager constructor(
+                            val subscriptionId = (it.data).in_app_subscription.subscription_id
+                            Log.i(TAG, "Subscription ID:  $subscriptionId")
+                            val subscriptionResult = (it.data).in_app_subscription
+                            if (subscriptionId.isEmpty()) {
+                                purchaseCallBack?.onSuccess(subscriptionResult, false)
+                            } else {
+                                purchaseCallBack?.onSuccess(subscriptionResult, true)
+                            }
                         } else {
-                            purchaseCallBack?.onSuccess(subscriptionResult, true)
+                            purchaseCallBack?.onError(CBException(ErrorDetail(message = GPErrorCode.PurchaseInvalid.errorMsg)))
                         }
-                    }else{
-                        purchaseCallBack?.onError(CBException(ErrorDetail(message = GPErrorCode.PurchaseInvalid.errorMsg)))
                     }
-                }
-                is ChargebeeResult.Error -> {
-                    Log.e(TAG, "Exception from Server - validateReceipt() :  ${it.exp.message}")
-                    purchaseCallBack?.onError(it.exp)
+                    is ChargebeeResult.Error -> {
+                        Log.e(TAG, "Exception from Server - validateReceipt() :  ${it.exp.message}")
+                        purchaseCallBack?.onError(it.exp)
+                    }
                 }
             }
-        }
-        }catch (exp: Exception){
+        } catch (exp: Exception) {
             Log.e(TAG, "Exception from Server- validateReceipt() :  ${exp.message}")
             purchaseCallBack?.onError(CBException(ErrorDetail(message = exp.message)))
         }
     }
 
-    fun queryAllPurchases(){
-        billingClient.queryPurchasesAsync(
-            BillingClient.SkuType.SUBS
-        ) { billingResult, activeSubsList ->
-            if (billingResult.responseCode == BillingClient.BillingResponseCode.OK) {
-                Log.i(TAG, "queryAllPurchases  :$activeSubsList")
+    private fun queryPurchaseHistoryFromStore(completionCallback: RestorePurchaseCallback) {
+        onConnected({ status ->
+            if (status) queryPurchaseHistory({ purchaseHistoryList ->
+                val storeTransactions = arrayListOf<PurchaseTransaction>()
+                storeTransactions.addAll(purchaseHistoryList)
+                CBRestorePurchaseManager.fetchStoreSubscriptionStatus(storeTransactions, completionCallback)
+            }, { error -> completionCallback.onError(error) })
+        }, { error ->
+            completionCallback.onError(error)
+        })
+    }
+
+    private fun queryPurchaseHistory(
+        completionCallback: (List<PurchaseTransaction>) -> Unit,
+        connectionError: (CBException) -> Unit
+    ) {
+        queryAllPurchaseHistory(CBPurchase.ProductType.SUBS.value, { subscriptionTransactionList ->
+            queryAllPurchaseHistory(
```

Comment:
If the `queryAllPurchaseHistory` is called only twice - first to query all Subs and next to query InApp, can we instead explicitly create two methods and call them sequentially? Should offer better readability/debugability.
```
queryAllSubsPurchaseHistory(...) 
queryAllInAppPurchaseHistory(..)
```

Reply:
Refactored the code here, please check and let me know if any changes. Thanks